src/poseidon.rs

```diff
@@ -0,0 +1,83 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+#![cfg(feature = "poseidon")]
+
+#[cfg(feature = "zk")]
+mod zk;
+
+mod zero_items;
+use zero_items::zero_items;
+
+use dusk_bls12_381::BlsScalar;
+use dusk_poseidon::sponge::hash as poseidon_hash;
+
+use crate::Aggregate;
+
+/// The Poseidon Merkle Tree
```

Comment:
I would encourage some more explanatory docs in this file.

Reply:
Totally agreed! I want to add detailed explanation for the unique structure with aggregation of the poseidon hash being implemented and the aggregation of the data left to the user. I created a new issue for this: #31